### PR TITLE
Tidy up the mock / live choice code in example app

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
- Represent “you only need to provide `key` and `clientId` if you’re using the live mode” using the type system.
- Simplify the client creation and state management (no need to create clients that we aren’t going to use).
- Make it clear that the view works with (well, will work with, once we no longer have to think about which features we’ve implemented) anything that conforms to the `ChatClient` interface.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new environment setting for chat client connections, allowing users to switch between mock and live modes.
	- Enhanced reaction animations with new visual effects, including rotation.

- **Bug Fixes**
	- Streamlined chat client initialization process for improved performance and usability.

### CODE REVIEW

File: Example/AblyChatExample/ContentView.swift [hunk 1]
```diff
@@ -2,6 +2,34 @@ import Ably
...
+    /// - Parameters:
+    ///   - key: Your Ably API key.
+    ///   - clientId: A string that identifies this client.
+    case live(key: String, clientId: String)
+
+    func createChatClient() -> ChatClient {
+        switch self {
+        case .mock:
+            return MockChatClient(
+                realtime: MockRealtime.create(),
```
Comment (maratal) [inline Example/AblyChatExample/ContentView.swift:19]: I know it's not from this PR, but `MockRealtime.create()` looks excessive in comparing to just `MockRealtime()`

File: Example/AblyChatExample/ContentView.swift [hunk 2]
```diff
@@ -2,6 +2,34 @@ import Ably
 import AblyChat
 import SwiftUI
 
+// Set mode to `.live` if you wish to connect to actual instances of the Chat client in either Prod or Sandbox environments. Setting the mode to `.mock` will use the `MockChatClient`, and therefore simulate all features of the Chat app.
+private let mode = Environment.mock
```
Comment (maratal) [inline Example/AblyChatExample/ContentView.swift:6]: Maybe put it inside `Environment` as static `current`?

**Review by maratal — COMMENTED** [review]
Left a few questions

File: Example/AblyChatExample/ContentView.swift (hunk 1 above)
Comment (lawrence-forooghian) [reply Example/AblyChatExample/ContentView.swift:19]: Did you see the comment on the `create()` method?

File: Example/AblyChatExample/ContentView.swift (hunk 2 above)
Comment (lawrence-forooghian) [reply Example/AblyChatExample/ContentView.swift]: Changed

**Review by maratal — APPROVED** [review]
Lgtm

**Pull request merged**